### ISSUE TITLE
Ideas for improving the documentation

### ISSUE THREAD

**Issue by mscheper** [reubano/csv2ofx#23]
I played with this library today, hoping it'd help me convert my financial data into a format I could upload to QuickBooks. Unfortunately, QuickBooks rejects the output files, with unhelpful error messages, even though GNUCash does them.

In any case, some improvement to the README.md documentation would've helped me understand this library a little more quickly. Here are some suggestions:

- Specify that the expected date format is _MM/DD/YYYY_—this is an unusual format (only two or three countries use it), so it ought to be made clear
- Mention that, by default, future-dated transactions are excluded (some of my transactions were missing because they were from the other side of the international dateline)
- Explain how `mapping` works. e.g. '`mapping` should be a map of output field names to functions that return values for those fields. The functions should expect a single parameter: a map of input field names to their values. For example, `{ 'date': lambda r: '%s/%s/%s' % (r['month'], r['day'], r['year']) }` expects the input CSV to have separate "month", "day" and "year" columns, and will combine these into a single "date" output column, in the _MM/DD/YYYY_ format.'
- For each field, particularly the optional ones, specify, which OFX elements they map to, e.g. that `notes` and `class` get combined into the `<MEMO>` element

**Comment by reubano** [comment]
Great ideas! This partially addressed in #22. As of the commit closing that issue, you can set a custom date format in the mapping. Also, one thing to point out is that the `mapping` values don't *have* to be functions.. something like `{'date': 'txn_date'}` is also acceptable. If you have time to submit a PR with necessary documentation updates, I would be very grateful!

**Comment by reubano** [comment]
I agree, I never really explained `split` in detail... it's [mentioned here](https://github.com/reubano/csv2ofx#optional-value-attributes) and is easiest to understand by comparing [`mint.csv`](https://github.com/reubano/csv2ofx/blob/master/data/test/mint.csv) (not split) to [`xero.csv`](https://github.com/reubano/csv2ofx/blob/master/data/test/xero.csv) (split). To parse `xero.csv` you would add `--collapse Description` to the command.

**Comment by reubano** [comment]
> Also, one thing to point out is that the mapping values don't have to be functions.. something like {'date': 'txn_date'}

Sorry, I misspoke here.... `{'bank': 'My Bank'}` is a better example. If you pass a string, that value is used for all rows. You only need a function if the value is dependent on information in the row. E.g., in the initial example, you could have `{'date': itemgetter('txn_date')}`.

**Comment by reubano** [comment]
> To parse xero.csv you would add `--collapse Description --mapping xero` to the command.

This works because [`xero.py`](https://github.com/reubano/csv2ofx/blob/master/csv2ofx/mappings/xero.py) has `mapping = { 'is_split': True, ...}`.

**Comment by reubano** [comment]
CR #16

**Comment by mscheper** [comment]
Thanks for the reply. I'll keep this tab open, but I'm trying to get some work finished before travelling so I honestly don't think it's likely I'll have time to make good improvements any time soon. But don't give up hope. ☺ Thanks for making the code available.
